backend/substrapp/tests/tests_misc.py

```diff
@@ -20,9 +27,39 @@ def write(self, s):
         pass
 
 
+class MockArchive:
+    def __init__(self, traversal=True):
+        if traversal:
+            self.files = ['../../foo.csv', '../bar.csv']
+        else:
+            self.files = ['foo.csv', 'bar.csv']
+
+    def __iter__(self):
+        return iter(self.files)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        return True
+
+    def namelist(self):
+        return self.files
+
+    def getnames(self):
+        return self.files
+
+
 class MiscTests(TestCase):
     """Misc tests"""
 
+    def setUp(self):
+        if not os.path.exists(DIRECTORY):
```

Comment:
You could use a temporary folder using the tempfile library (`/tmp` might not be available and it would be safer): https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp ?

Reply:
Yes, but other test use also `/tmp/xxx` like  `tests_task.py`.
But I don't use it in the test so I will remove it :)

Reply:
Makes sense.